share/eds/cache/cache.go

```diff
@@ -0,0 +1,49 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/filecoin-project/dagstore"
+	"github.com/filecoin-project/dagstore/shard"
+	logging "github.com/ipfs/go-log/v2"
+	"go.opentelemetry.io/otel"
+)
+
+var (
+	log   = logging.Logger("share/eds/cache")
+	meter = otel.Meter("eds_store_cache")
+)
+
+var (
+	ErrCacheMiss = errors.New("accessor not found in blockstore cache")
+)
+
+// Cache is an interface that defines the basic Cache operations.
+type Cache interface {
+	// Get retrieves an item from the Cache.
+	Get(shard.Key) (Accessor, error)
+
+	// GetOrLoad attempts to get an item from the Cache and, if not found, invokes
+	// the provided loader function to load it into the Cache.
+	GetOrLoad(
+		ctx context.Context,
+		key shard.Key,
+		loader func(context.Context, shard.Key) (Accessor, error),
+	) (Accessor, error)
+
+	// Remove removes an item from Cache.
+	Remove(shard.Key) error
+
+	// EnableMetrics enables metrics in Cache
+	EnableMetrics() error
+}
+
+// Accessor is a interface type returned by cache, that allows to read raw data by reader or create
+// readblockstore
+type Accessor interface {
+	Blockstore() (dagstore.ReadBlockstore, error)
+	Reader() io.Reader
+	io.Closer
+}
```

Comment:
Accessor interface should be defined in dagstore. It is part of its api, that is implemented by dagstore.Accessor type. For some reason they use exact type instead of interface, so it is impossible to have a custom implementation of it and it such ability is crucial for this PR to be possible.
Ideally I would make PR to our dagstore fork to use interface, but it feels like pointless work at this point, given an estimate, that we will get rid of it soon with ipldv2 work going on. And if it is not in dagstore, I don't know where it could be moved and not cause dependency graph issues. So it lives here for now.

Reply:
Agree with reasoning to ignore this